### PR TITLE
add random gun safe spawners

### PR DESCRIPTION
**Pull request by MilonPL**
## About the PR

Someone mentioned it in le server meeting, so I decided to add it in. Also adds a filled safe for the M-90 as that's already a thing on Shoko so it's probably fine. The handgun safe has a low chance of spawning with universals, WT550s or miniature energy guns. Miniature guns were also on sub so I assume that's fine too.

## Why / Balance

Gives security a more varied choice of weapons and adds a random element to the rounds. Kinda cool.

## Media

obraz_2024-07-07_020656080

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Added randomized handgun, rifle and shotgun safe spawners for mapping.

### CODE REVIEW

File: Resources/Prototypes/DeltaV/Catalog/Fills/Lockers/security.yml
Comment (Colin-Tel) [inline Resources/Prototypes/DeltaV/Catalog/Fills/Lockers/security.yml:1]: Please un-touch the whitespace changes in this file.

**Review by Colin-Tel — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/DeltaV/Catalog/Fills/Lockers/security.yml [hunk 1]
```diff
@@ -65,3 +65,15 @@
     contents:
     - id: WeaponEnergyGunMini
       amount: 3
+
+- type: entity
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Catalog/Fills/Lockers/security.yml:69]: this should either be upstreamed or in its own file, and what colin said

File: Resources/Prototypes/DeltaV/Entities/Markers/Spawners/Random/safes.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+- type: entity
+  name: random shotgun safe spawner
+  id: ShotgunSafeSpawner
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Markers/Spawners/Random/safes.yml]: make a base gun safe spawner that has sprite + offset: 0 that they all copy pasted

also format with style guide i think we just use upstreams

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/DeltaV/Catalog/Fills/Lockers/security.yml (hunk 1 above)
Comment (MilonPL) [reply Resources/Prototypes/DeltaV/Catalog/Fills/Lockers/security.yml:69]: I mean the M90 is a DeltaV gun, isn't it supposed to be with the rest of DeltaV gun safe fills?

Comment (deltanedas) [reply Resources/Prototypes/DeltaV/Catalog/Fills/Lockers/security.yml:69]: oh i cant read

Comment (MilonPL) [reply Resources/Prototypes/DeltaV/Catalog/Fills/Lockers/security.yml:69]: obraz_2024-07-07_135800440

File: Resources/Prototypes/DeltaV/Entities/Markers/Spawners/Random/safes.yml [hunk 3]
```diff
@@ -1,55 +1,52 @@
...
     - type: Sprite
       layers:
         - state: red
         - sprite: Structures/Storage/closet.rsi
           state: shotguncase
+    - type: RandomSpawner
+      offset: 0
+
+- type: entity
+  name: random safe spawner
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Markers/Spawners/Random/safes.yml]: pro put name on the abstract one

Comment (MilonPL) [reply Resources/Prototypes/DeltaV/Entities/Markers/Spawners/Random/safes.yml]: that's the shotgun safe spawner not the abstract one

Comment (MilonPL) [reply Resources/Prototypes/DeltaV/Entities/Markers/Spawners/Random/safes.yml]: ohh nvm i think i might be dumb sometimes or most of the times

**Review by deltanedas — APPROVED** [review]

**Review by Colin-Tel — APPROVED** [review]

**Pull request merged**